### ISSUE TITLE
[Bug] install with pip will reinstall cpu version of torch

### ISSUE THREAD

**Issue by kaiwang0112006** [InternLM/lmdeploy#3549]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.

### Describe the bug

when I install lmdeploy-0.8.0+cu118-cp311-cp311-manylinux2014_x86_64.whl with pip, it will try to download and reinstall cpu version of torch, although I have torch==2.3.0+cu121 installed on my envirment.

### Reproduction

    pip install lmdeploy-0.8.0+cu118-cp311-cp311-manylinux2014_x86_64.whl

result:

    ...
    Requirement already satisfied: charset-normalizer<4,>=2 in /opt/anaconda3/lib/python3.11/site-packages (from requests->outlines->lmdeploy==0.8.0+cu118) (2.0.4)
    Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/anaconda3/lib/python3.11/site-packages (from requests->outlines->lmdeploy==0.8.0+cu118) (2.0.7)
    Requirement already satisfied: MarkupSafe>=2.0 in /opt/anaconda3/lib/python3.11/site-packages (from jinja2->torch<=2.6.0,>=2.0.0->lmdeploy==0.8.0+cu118) (2.1.3)
    Requirement already satisfied: attrs>=22.2.0 in /opt/anaconda3/lib/python3.11/site-packages (from jsonschema->outlines->lmdeploy==0.8.0+cu118) (23.1.0)
    Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /opt/anaconda3/lib/python3.11/site-packages (from jsonschema->outlines->lmdeploy==0.8.0+cu118) (2023.7.1)
    Requirement already satisfied: rpds-py>=0.7.1 in /opt/anaconda3/lib/python3.11/site-packages (from jsonschema->outlines->lmdeploy==0.8.0+cu118) (0.10.6)
    Requirement already satisfied: markdown-it-py<3.0.0,>=2.2.0 in /opt/anaconda3/lib/python3.11/site-packages (from rich->mmengine-lite->lmdeploy==0.8.0+cu118) (2.2.0)
    Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /opt/anaconda3/lib/python3.11/site-packages (from rich->mmengine-lite->lmdeploy==0.8.0+cu118) (2.15.1)
    Requirement already satisfied: mdurl~=0.1 in /opt/anaconda3/lib/python3.11/site-packages (from markdown-it-py<3.0.0,>=2.2.0->rich->mmengine-lite->lmdeploy==0.8.0+cu118) (0.1.0)
    Using cached peft-0.14.0-py3-none-any.whl (374 kB)
    Using cached pydantic-2.11.4-py3-none-any.whl (443 kB)
    Using cached pydantic_core-2.33.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)
    Downloading torch-2.6.0-cp311-cp311-manylinux1_x86_64.whl (766.7 MB)
       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.8/766.7 MB 227.0 kB/s eta 0:56:15

although:

    python -c "import torch; print(torch.__version__)"
    2.3.0+cu121

### Environment

```Shell
Python 3.11.7
torch==2.3.0+cu121
```

### Error traceback

```Shell

```

**Comment by grimoire** [comment]
install with `--no-deps` can skip all dependency.
And you can install the dependency [here](https://github.com/InternLM/lmdeploy/blob/main/requirements/runtime_cuda.txt) manually.

**Comment by kaiwang0112006** [comment]
ok
